packages/firebase/index.d.ts

```diff
@@ -95,6 +104,49 @@ declare namespace firebase {
     variant?: string
   ): void;
 
+  /**
+   * Sets log level for all Firebase packages.
+   *
+   * All of the log types above the current log level will be captured (i.e. if
+   * you set the log level to `info`, errors will still be logged, but `debug` and
+   * `verbose` logs will not)
+   */
+  function setLogLevel(logLevel: LogLevel): void;
+
+  /**
+   * Sets log handler for all Firebase packages.
+   * @param logCallback An optional custom log handler that will execute user code whenever
+   * the Firebase SDK makes a logging call.
+   */
+  function onLog(
+    logCallback: (callbackParams: {
+      /**
+       * Level of event logged.
+       */
+      level: LogLevel;
+      /**
+       * Any text from logged arguments joined into one string.
+       */
+      message: string;
+      /**
+       * The raw arguments passed to the log call.
+       */
+      args: unknown[];
+      /**
+       * A string indicating the source of the log call, usually a package name
```

Comment:
Is it not always a package name? What are the exceptions ?

Reply:
There aren't any exceptions I can think of but I wanted to be safe in case we log a message from some util or pass through an error from a dep or something?

Reply:
I feel we should just say it will be a package name since it's the case at the moment.

Reply:
Done.